src/swap/useSwapQuote.ts

```diff
@@ -70,25 +282,38 @@ const useSwapQuote = () => {
       const swapPrice = useGuaranteedPrice
         ? quote.unvalidatedSwapTransaction.guaranteedPrice
         : quote.unvalidatedSwapTransaction.price
+      const price =
+        updatedField === Field.FROM
+          ? swapPrice
+          : new BigNumber(1).div(new BigNumber(swapPrice)).toFixed()
       const estimatedPriceImpact = quote.unvalidatedSwapTransaction.estimatedPriceImpact
-      const updatedExchangeRate: ExchangeRate = {
+      const quoteResult: QuoteResult = {
         toTokenAddress: toToken.address,
         fromTokenAddress: fromToken.address,
         swapAmount: swapAmount[updatedField],
-        price:
-          updatedField === Field.FROM
-            ? swapPrice
-            : new BigNumber(1).div(new BigNumber(swapPrice)).toFixed(),
+        price,
+
         provider: quote.details.swapProvider,
         estimatedPriceImpact: estimatedPriceImpact
           ? new BigNumber(estimatedPriceImpact).dividedBy(100)
           : null,
       }
 
-      return updatedExchangeRate
+      if (shouldPrepareTransactions) {
```

Comment:
I agree with this, however here it's a temporary adjustment until we remove the feature flag.
This isn't meant to stay like this. I do think it's the right place to have it to avoid duplicating the hook.
Maybe not ideal, but the alternative I was seeing didn't make sense in this context.

Reply:
ok, maybe it would be helpful to leave a comment to that effect?